src/appleseed/foundation/meta/tests/test_qmc.cpp

```diff
@@ -549,8 +550,9 @@ TEST_SUITE(Foundation_Math_QMC)
             for (size_t d = 0; d < Dimension; ++d)
             {
                 const double s = radical_inverse<double>(Primes[d], i);
-
-                fprintf(file, "%.16f", s);
+                
+                std::string fmt = "%." + std::to_string(std::numeric_limits<double>::max_digits10) + "f";
```

Comment:
Even if we're in a unit test, the multiple memory allocations (to create the multiple strings, probably at least 3) for each SampleCount * Dimension is a bit too much on the heavy side.

I think we can live with `"%.16f"` here since we're generating values, not round-tripping from/to a file.

Reply:
Or `"%.17f"` while we're at it :)

Reply:
Done.